dokka-extensions/src/main/kotlin/io/spine/tools/dokka/plugin/ExcludeInternalTransformer.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2022, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.dokka.plugin
+
+import io.spine.annotation.Internal
+import org.jetbrains.dokka.base.transformers.documentables.SuppressedByConditionDocumentableFilterTransformer
+import org.jetbrains.dokka.model.Annotations
+import org.jetbrains.dokka.model.Annotations.Annotation
+import org.jetbrains.dokka.model.Documentable
+import org.jetbrains.dokka.model.properties.WithExtraProperties
+import org.jetbrains.dokka.plugability.DokkaContext
+
+/**
+ * Excludes everything annotated with [Internal] from the resulting documentation.
+ *
+ * Works for packages, types, fields and methods.
+ */
+public class ExcludeInternalTransformer(dokkaContext: DokkaContext) :
+    SuppressedByConditionDocumentableFilterTransformer(dokkaContext) {
+
+    /**
+     * Tells whether the passed [Documentable] should be excluded from the documentation.
+     *
+     * Returns `true` if the passed piece is annotated with the [Internal] annotation. Otherwise,
+     * returns `false`.
+     */
+    override fun shouldBeSuppressed(d: Documentable): Boolean {
+        /**
+         * [WithExtraProperties] is a Dokka-interface that adds a map-field for extra properties.
```

Comment:
Usually, we don't have block comments for the in-line expressions, other than to comment suppressions.

There are some exceptions, but this is not the one.

If you want to comment on the implementation details, please move this text to the method-level docs. There is an `@implNote` tag in Javadocs for that. AFAIK, KDoc doesn't allow that. In any case, the explanation may be just the continuation of the documentation you already have in place for the method.